### ISSUE TITLE
Support for "or" and "not" in tag queries

### ISSUE THREAD

**Issue by Remich** [srid/neuron#358]
Hi,

I figured by trial and error (_missing in documentation_) that automatic links support the "and-operator": e.g.: `[[z:zettels?tag=science&tag=math&timeline]]`

It would also be nice to have support for "or" and "not" operators, and maybe parenthesis, in order to create arbitrarily complex search queries for auto-linking.

This would allow for a better auto-linking experience.

**Comment by srid** [comment]
We could add a `q` parameter to support complex queries, for example:

```
[[z:zettels?q="(tag:science OR tag:physics) -(tag:math)"]]
```

Which in flexible links of #312 would look like:

```
{{zettels | q = (tag:science OR tag:physics) -(tag:math) }}
```

**Comment by srid** [comment]
FTR, `and` / `or` differentiation is available in the code via the `TagQuery` type as of #500. That type will have to be enriched further in order to support this feature (at this point, however, the whole query thing must refactored into a plugin of its own).

**Comment by srid** [comment]
This should get implemented after #407

**Comment by s9gf4ult** [comment]
I would 

```
{{zettels | q = (tag.science | tag.physics) & !(tag.math) }}
```

* `&` for AND
* `|` for OR 
* `!` for NOT
* `()` for brackets

Also, I would allow to query by arbitrary zettel's metadata. 

Consider this meta

``` yaml
issue:
  status: "bug"
```

You could query it by

```
{{ zettels | q = issue.status = "bug" }}
```

So, there must be also

* `"` for string literals
* `true` `false` for boolean literals
* `=` for comparison of two metadata values

**Comment by srid** [comment]
Don't see the value of favouring C-like operators instead of using standard search query operators used by other apps and sites, but doing the below is a good idea.

>  I would allow to query by arbitrary zettel's metadata

**Comment by srid** [comment]
Closing per https://neuron.zettel.page/next ... to further discuss this idea, post in https://github.com/srid/emanote/discussions/48
